src/lib.rs

```diff
@@ -452,12 +468,21 @@ impl Client {
             }
         };
 
+        let mut default_tags = options.default_tags;
+        // Prepending default tags in order to let the user be able to override system tags with custom
+        // default tags
+        default_tags.extend(Options::default_system_tags());
+        default_tags.sort_by(|(key_a, _), (key_b, _)| key_a.cmp(key_b));
+        default_tags.dedup_by(|(key_a, _), (key_b, _)| key_a == key_b);
```

Comment:
I would add a few comments, since this section doesn't seem obvious to me

```suggestion
        // Append the system tags so they get overwritten by custom tags set by the user
        default_tags.extend(Options::default_system_tags());
        // `sort_by` is guaranteed to keep the order between equal elements, so the custom tags will always end up at the front, before system tags
        default_tags.sort_by(|(key_a, _), (key_b, _)| key_a.cmp(key_b));
        // `dedup_by` keeps the first occurrence of an element
        default_tags.dedup_by(|(key_a, _), (key_b, _)| key_a == key_b);
```